### ISSUE TITLE
Date is interpreted as an object in diff-array.js

### ISSUE THREAD

**Issue by bjarketrux** [Urigo/angular-meteor#418]
I have a problem of a date not being updated on the database when a client changes the value (value is updated on client).

The code in `diff-array.js` looks like it interprets dates as an objects, and tries to compare all properties on it, but there isn't any.

```
// Can tell whether a value is an object and not an array
var isActualObject = function (value) {
  return angular.isObject(value) && !angular.isArray(value);
};
```

If you call the above code with a `new Date()` as the value it will return true.
Is this a bug or intended behavior?

Thanks

**Comment by Urigo** [comment]
@bjarketrux do you think you can add a failed test that catches this scenario?

**Comment by bjarketrux** [comment]
First thing tomorrow!

**Comment by bjarketrux** [comment]
Fixed in pull request #422
